src/vs/workbench/services/extensions/common/extensionsProposedApi.ts

```diff
@@ -54,16 +54,13 @@ export class ExtensionsProposedApi {
 		}
 	}
 
-	private doUpdateEnabledApiProposals(_extension: IExtensionDescription): void {
+	private doUpdateEnabledApiProposals(extension: IExtensionDescription): void {
 
-		// this is a trick to make the extension description writeable...
-		type Writeable<T> = { -readonly [P in keyof T]: Writeable<T[P]> };
-		const extension = <Writeable<IExtensionDescription>>_extension;
-		const key = ExtensionIdentifier.toKey(_extension.identifier);
+		const key = ExtensionIdentifier.toKey(extension.identifier);
 
 		// warn about invalid proposal and remove them from the list
-		if (isNonEmptyArray(extension.enabledApiProposals)) {
-			extension.enabledApiProposals = extension.enabledApiProposals.filter(name => {
+		if (isNonEmptyArray(extension.enabledApiProposalNames)) {
+			extension.enabledApiProposalNames = extension.enabledApiProposalNames.filter(name => {
```

Comment:
We now have two lists of enabled proposals and they now diverge. Only the latter `enabledApiProposalNames` is corrected with product-json information and the former is left as authored (untrusted). We should not have this, I would recommended to stick to `enabledApiProposals` and make consumers do the parsing